staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -134,15 +168,51 @@ func StartController(ctx context.Context, kubeClient kubernetes.Interface, drive
 		defer logger.V(3).Info("Stopping")
 		c.run(ctx)
 	}()
-
-	// Sync each pool once.
-	for poolName := range resources.Pools {
-		c.queue.Add(poolName)
-	}
-
 	return c, nil
 }
 
+// Options contains various optional settings for [StartController].
+type Options struct {
+	// DriverName is the required name of the DRA driver.
+	DriverName string
+
+	// KubeClient is used to read Node objects (if necessary) and to access
+	// ResourceSlices. It must be specified.
+	KubeClient kubernetes.Interface
+
+	// If the owner is a v1.Node, then the NodeName field in the
+	// ResourceSlice objects is set and used to identify objects
+	// managed by the controller. The UID is not needed in that
+	// case, the controller will determine it automatically.
+	//
+	// The owner must be cluster-scoped. This is not always possible,
+	// therefore it is optional. A driver without a owner must take
+	// care that remaining slices get deleted manually as part of
+	// a driver uninstall because garbage collection won't work.
+	Owner *Owner
+
+	// This is the initial desired set of slices. As with
+	// [Controller.Update], the controller takes ownership of the resources
+	// instance.  The content must not get modified by the caller.
```

Comment:
Is there a particular reason not to do a `DeepCopy()` in this controller to prevent drivers from accidentally modifying this object / remove the need for them to do the `DeepCopy()` themselves? I know I got this wrong the first time I implemented calls to this API and I'm sure I won't be the last.

Reply:
Just the cost. If the caller creates a new instance, then the DeepCopy would be redundant.

But you are right, it's probably "better safe than sorry". I'll add a DeepCopy here.